tests/deque_test.go

```diff
@@ -0,0 +1,406 @@
+package tests
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+)
+
+var randGenerator *rand.Rand
+var runes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_!@#$%^&*()-=+[]\\;':,.<>/?~.|")
+
+var (
+	deqNormalValues []string
+	deqEdgeValues   []string
+)
+
+func randStr(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = runes[randGenerator.Intn(len(runes))]
+	}
+	return string(b)
+}
+
+func init() {
```

Comment:
Rename to initDeque to avoid generic names in the test namespace. You can call the initDeque method at the start of each test function.